.pre-commit-config.yaml

```diff
@@ -3,25 +3,25 @@
 
 repos:
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.15.2
+    rev: v3.19.1
     hooks:
       - id: pyupgrade
         args:
-          - --py38-plus
-  - repo: https://github.com/pycqa/flake8
-    rev: 7.0.0
-    hooks:
-      - id: flake8
-        language_version: python3
+          - --py39-plus
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.4.0
+    rev: v5.0.0
     hooks:
       - id: trailing-whitespace
       - id: end-of-file-fixer
         exclude: tests/.*\.json
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.4.9
+    rev: v0.9.5
     hooks:
       - id: ruff
         args: ["--fix"]
       - id: ruff-format
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.15.0
+    hooks:
+      - id: mypy
+        additional_dependencies: [types-requests, types-python-dateutil]
```

Comment:
I strongly disagree with putting mypy in pre-commit because it means that either:

- you have no dependencies installed when mypy is running, so mypy misses any type errors that come from usage of dependencies
- you have to entirely duplicate your dependency management from your core `pyproject.toml`
- (this case, with incomplete dependencies here) you get two different results from mypy: one when running in pre-commit and one when running with all dependencies. And then contributors get confused why mypy passes on pre-commit but still fails on CI.